code/game/objects/structures/holosign.dm

```diff
@@ -52,37 +59,102 @@
+
 /obj/structure/holosign/wetsign
 	name = "wet floor sign"
+	icon = 'icons/effects/effects.dmi' //spinny sign cooler
 	desc = "The words flicker as if they mean nothing."
-	icon = 'icons/effects/effects.dmi'
 	icon_state = "holosign"
 
 /obj/structure/holosign/barrier
-	name = "holobarrier"
-	desc = "A short holographic barrier which can only be passed by walking."
+	name = "security holobarrier"
+	desc = "A strong short security holographic barrier used for crowd control and blocking crime scenes. Can only be passed by walking."
 	icon_state = "holosign_sec"
+	base_icon_state = "holosign_sec"
 	pass_flags_self = PASSTABLE | PASSGRILLE | PASSGLASS | LETPASSTHROW
 	density = TRUE
 	max_integrity = 20
-	var/allow_walk = TRUE //can we pass through it on walk intent
+	COOLDOWN_DECLARE(cooldown_open)
+	///Can we pass through it on walk intent?
+	var/allow_walk = TRUE
+	///Can it be temporarily opened with the holosign projector?
+	var/openable = TRUE
+	///Is it opened?
+	var/opened = FALSE
+	///What is the icon of opened holobarrier?
+	var/pass_icon_state = "holosign_pass"
 
 /obj/structure/holosign/barrier/CanAllowThrough(atom/movable/mover, border_dir)
 	. = ..()
 	if(.)
 		return
+	if(opened)
+		return TRUE
 	if(iscarbon(mover))
-		var/mob/living/carbon/C = mover
-		if(C.stat) // Lets not prevent dragging unconscious/dead people.
+		var/mob/living/carbon/moving_carbon = mover
+		if(moving_carbon.stat) // Lets not prevent dragging unconscious/dead people.
 			return TRUE
-		if(allow_walk && C.m_intent == MOVE_INTENT_WALK)
+		if(allow_walk && moving_carbon.m_intent == MOVE_INTENT_WALK)
 			return TRUE
 
+// /obj/structure/holosign/barrier/ranged_item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+// 	. = ..()
+// 	if(tool != projector)
+// 		return
+// 	if(openable)
+// 		open(user)
+
+/obj/structure/holosign/barrier/attackby(obj/item/tool, mob/living/user, params)
+	. = ..()
+	if(tool != projector)
+		return
+	qdel(src)
+
+/obj/structure/holosign/barrier/update_icon_state()
+	if(openable)
+		if(!opened)
+			icon_state = base_icon_state
+		else
+			icon_state = pass_icon_state
+	if(use_vis_overlay)
+		SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
+		var/turf/our_turf = get_turf(src)
+		SSvis_overlays.add_vis_overlay(src, icon, icon_state, ABOVE_MOB_LAYER, MUTATE_PLANE(GAME_PLANE_UPPER, our_turf), dir) //you see mobs under it, but you hit them like they are above it
+	. = ..()
+
+/obj/structure/holosign/barrier/proc/open(user)
+	if(!openable)
+		balloon_alert(user, "unable!")
+		return
+
+	if(!COOLDOWN_FINISHED(src, cooldown_open))
+		balloon_alert(user, "on cooldown!")
+		return
+
+	if(!opened)
+		density = FALSE
+		opened = TRUE
+		playsound(src, 'sound/machines/door_open.ogg', 50, TRUE)
+	else
+		density = TRUE
+		opened = FALSE
+		playsound(src, 'sound/machines/door_close.ogg', 50, TRUE)
```

Comment:
This could be shortened a bit, tho x = !x syntax bit me in the ass before so you'd have to test it locally if it works
```suggestion
	density = !density
	opened = !opened
	if(!opened)
		playsound(src, 'sound/machines/door_open.ogg', 50, TRUE)
	else
		playsound(src, 'sound/machines/door_close.ogg', 50, TRUE)
```